tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -1379,6 +1380,22 @@ void SpirvBuilder::decorateCoherent(SpirvInstruction *target,
   mod->addDecoration(decor);
 }
 
+void SpirvBuilder::decorateLinkage(SpirvInstruction *targetInst,
+                                   SpirvFunction *targetFunc,
+                                   llvm::StringRef name,
+                                   spv::LinkageType linkageType,
+                                   SourceLocation srcLoc) {
```

Comment:
I'd add `assert((targetInst == nullptr) != (targetFunc == nullptr));` and explain we have to set a decoration for the linkage of a global variable or a function, but we cannot set them at the same time.

Note that `(targetInst == nullptr) != (targetFunc == nullptr)` is the same as `(targetInst != nullptr || targetFunc != nullptr) && (targetInst == nullptr || targetFunc == nullptr)`